srml/executive/src/lib.rs

```diff
@@ -311,24 +311,23 @@ impl<
 			}
 
 			// check index
-			let mut expected_index = <system::Module<System>>::account_nonce(sender);
+			let expected_index = <system::Module<System>>::account_nonce(sender);
 			if index < &expected_index {
 				return TransactionValidity::Invalid(ApplyError::Stale as i8)
 			}
-			if *index > expected_index + As::sa(256) {
-				return TransactionValidity::Unknown(ApplyError::Future as i8)
-			}
 
-			let mut deps = Vec::new();
-			while expected_index < *index {
-				deps.push((sender, expected_index).encode());
-				expected_index = expected_index + One::one();
-			}
+			let index = *index;
+			let provides = vec![(sender, index).encode()];
+			let requires = if expected_index < index {
```

Comment:
This vector should really be an `Option`, since it now has only 0 or 1 element.

Reply:
We can't change it to `Option` - `TransactionValidity` is a generic type that might be used by many different runtimes, some of them might need multiple `required` tags (UTXO for instance).